jwst/cube_build/instrument_defaults.py

```diff
@@ -42,887 +41,1406 @@ def __init__(self):
+        """
+        weight_power = self.Info[parameter1][parameter2]["msm_power"]
         return weight_power
 
-    def GetSoftRad(self, parameter1, parameter2):
-        softrad = self.Info[parameter1][parameter2]['softrad']
+    def get_soft_rad(self, parameter1, parameter2):
+        """
+        Get modified shepherd soft_rad value defined by given parameters.
+
+        Parameters
+        ----------
+        parameter1 : str
+            Grating for NIRSpec data/ channel for MIRI data
+        parameter2 : str
+            Filter for NIRSpec data/ band for MIRI data
+
+        Returns
+        -------
+        softrad : float
+            Modified shepherd weighting parameter for weight type = msm
+        """
+        softrad = self.Info[parameter1][parameter2]["softrad"]
         return softrad
 
-    def GetScaleRad(self, parameter1, parameter2):
-        scalerad = self.Info[parameter1][parameter2]['scalerad']
+    def get_scale_rad(self, parameter1, parameter2):
+        """
+        Get modified shepherd scale_rad value define by given parameters.
+
+        Parameters
+        ----------
+        parameter1 : str
+            Grating for NIRSpec data/ channel for MIRI data
+        parameter2 : str
+            Filter for NIRSpec data/ band for MIRI data
+
+        Returns
+        -------
+        scale_rad : float
+            Modified shepherd scale_rad value
+        """
+        scalerad = self.Info[parameter1][parameter2]["scalerad"]
         return scalerad
 
-    def GetScale(self, parameter1, parameter2):
-        scale = (self.Info[parameter1][parameter2]['ascale'],
-                 self.Info[parameter1][parameter2]['bscale'],
-                 self.Info[parameter1][parameter2]['wscale'])
+    def get_scale(self, parameter1, parameter2):
+        """
+        Get spatial and spectral scales for  for data defined by  given parameters.
+
+        Parameters
+        ----------
+        parameter1 : str
+            Grating for NIRSpec data/ channel for MIRI data
+        parameter2 : str
+            Filter for NIRSpec data/ band for MIRI data
+
+        Returns
+        -------
+        scale : (float, float, float)
+            Spatial scale, spatial scale, and spectral scale
+        """
+        scale = (
+            self.Info[parameter1][parameter2]["ascale"],
+            self.Info[parameter1][parameter2]["bscale"],
+            self.Info[parameter1][parameter2]["wscale"],
+        )
         return scale
 
-    def Get_multichannel_table(self, weighting):
-        table = (self.multich_wavelength,
-                 self.multich_sroi,
-                 self.multich_wroi,
-                 self.multich_power,
-                 self.multich_softrad,
-                 self.multich_scalerad)
+    def get_multichannel_table(self):
+        """
+        Get the MRS multichannel table containing flux weighting parameters.
+
+        Returns
+        -------
+        table : dictionary
```

Comment:
Parameters should be documented with their actual Python types whenever possible, so use the type `dict` here, instead of the word "dictionary".
```suggestion
        table : dict
```